g2o/core/base_constant_edge.h

```diff
@@ -0,0 +1,185 @@
+// This code is based on g2o/core/base_binary_edge.h and
+// g2o/core/base_multi_edge.h
+//
+// g2o - General Graph Optimization
+// Copyright (C) 2011 R. Kuemmerle, G. Grisetti, H. Strasdat, W. Burgard
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+//
+// * Redistributions of source code must retain the above copyright notice,
+//   this list of conditions and the following disclaimer.
+// * Redistributions in binary form must reproduce the above copyright
+//   notice, this list of conditions and the following disclaimer in the
+//   documentation and/or other materials provided with the distribution.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+// IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED
+// TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A
+// PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+// HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED
+// TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+#ifndef G2O_BASE_CONSTANT_EDGE_H
+#define G2O_BASE_CONSTANT_EDGE_H
+
+
+#include <iostream>
+#include <limits>
+
+#include "base_edge.h"
+#include "robust_kernel.h"
+#include "g2o/config.h"
+#include "g2o/stuff/misc.h"
+
+namespace g2o {
+
+  namespace {
+    // assumes i < j
+    constexpr int pair_to_index(const int i, const int j)
+    {
+      return j * (j - 1) / 2 + i;
+    }
+    constexpr std::pair<int, int> index_to_pair(const int k)
+    {
+      const int j = std::sqrt(2*(k+1))+.5;
+      const int i = k - j * (j - 1) / 2;
+      return {i, j};
+    }
+
+    template<int N, typename... Types>
+    using NthType = typename std::tuple_element<N, std::tuple<Types...>>::type;
```

Comment:
Since you're just using this once, you might just want to define it in your class so you're not adding this name is the g2o namespace (same goes for most of these other helpers)

Reply:
Agreed, I moved it in.